src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -536,6 +537,37 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
     }
 
     /// <summary>
+    /// This converter is used to attempt to convert TypedParameter into a json object
+    /// </summary>
+    public class TypedParameterConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(TypedParameter);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var obj = JValue.Load(reader);
+            return new TypedParameter(obj["Name"].ToString(), obj["TypeName"].ToString(), (int)obj["TypeRank"], obj["DefaultValue"].ToString());
```

Comment:
is it possible that one of these parameters is null - should we default to something?

Reply:
Yes, they are defaulted in the constructor

Reply:
what I mean is, if you call ToString() on null does an exception get thrown?

Reply:
https://stackoverflow.com/questions/11446838/why-does-tostring-on-a-null-string-cause-a-null-error-when-tostring-works

Reply:
not sure what would happen here :(

Reply:
Great question, let me do some testing

Reply:
@mjkkirschner It turns out [JsonConstructor] has higher priority than the `ReadJson()` here so this is not called at all :) I will remove these two lines and throw not implmented exception to avoid confusion.